geopandas/geodataframe.py

```diff
@@ -133,6 +133,8 @@ class GeoDataFrame(GeoPandasBase, DataFrame):
 
     def __init__(self, data=None, *args, geometry=None, crs=None, **kwargs):
         with compat.ignore_shapely2_warnings():
+            if kwargs.get("copy") is None and isinstance(data, DataFrame):
```

Comment:
You need to check if it is a pandas dataframe but not a GeoPandas geodataframe. GeoDataFrame is a subclass, so this will return True.

Reply:
>  GeoDataFrame is a subclass, so this will return True.

ah yes, that's my bad.